prototype/src/js/AlphaOrienter.jsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import {
+    alphaConstants,
+    cssOffsetPropertyName,
+    hideOrienterDelay
+} from './constants';
+import { delay, cancelDelay, calculateNormalizedAlphaOffset } from './utils';
+import DeviceSVG from './svg/DeviceSVG';
+import TargetSVG from './svg/TargetSVG';
+import ArrowLeftSVG from './svg/ArrowLeftSVG';
+import ArrowRightSVG from './svg/ArrowRightSVG';
+
+const classNames = require('classnames');
+
+export default class AlphaOrienter extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            offset: 1
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.setPosition(nextProps.alpha);
+        this.hideIfDone();
+    }
+
+    hideIfDone() {
+        if (!this.delay && this.state.offset === 0) {
+            this.delay = delay(hideOrienterDelay, () => {
+                this.props.onOrientationComplete();
+            });
+        } else if (this.delay && this.state.offset !== 0) {
+            cancelDelay(this.delay);
+            this.delay = null;
+        }
+    }
+
+    setPosition(reading) {
+        let offset = this.calculateOffset(reading);
+
+        this.setState({ offset: offset });
+
+        this.deviceEl.style.setProperty(cssOffsetPropertyName, -offset + 'px');
+        this.targetEl.style.setProperty(cssOffsetPropertyName, offset + 'px');
+        this.arrowEl &&
+            this.arrowEl.style.setProperty(
+                cssOffsetPropertyName,
+                -offset + 'px'
+            );
+    }
+
+    calculateOffset(reading) {
+        const normalizedOffset = calculateNormalizedAlphaOffset(
+            this.props.target,
+            reading
+        );
+        const containerSize = this.containerEl.offsetWidth;
+        const size = this.deviceEl.offsetWidth;
+        const maxOffset = (containerSize - size) / 2;
+        const sign = normalizedOffset < 0 ? -1 : 1;
+        return sign * Math.sqrt(sign * normalizedOffset) * maxOffset;
+    }
+
+    render() {
+        const { offset } = this.state;
+
+        let headingText, arrowSVG;
+        if (offset < 0) {
+            headingText = 'Turn left until the boxes align';
+            arrowSVG = <ArrowLeftSVG />;
+        } else if (offset > 0) {
+            headingText = 'Turn right until the boxes align';
+            arrowSVG = <ArrowRightSVG />;
+        } else {
+            headingText = 'Nice!';
+            arrowSVG = null;
+        }
+
+        const deviceClassName = classNames('illustration -device', {
```

Comment:
Seems like this `classnames` library might be useful to include in other Civic Apps React projects.

Reply:
Super useful.